libafl/src/fuzzer/mod.rs

```diff
@@ -420,43 +420,79 @@ where
     ) -> Result<(ExecuteInputResult, Option<CorpusId>), Error>
     where
         EM: EventFirer<State = Self::State>,
+        OT: ObserversTuple<Self::State> + Serialize,
     {
         let exec_res = self.check_results(state, manager, &input, observers, exit_kind)?;
-        let corpus_id = self.process_execution(
-            state,
-            manager,
-            input,
-            &exec_res,
-            observers,
-            exit_kind,
-            send_events,
-        )?;
+        let corpus_id = self.process_execution(state, manager, &input, &exec_res, observers)?;
+
+        // Now send off the event
+        match exec_res {
```

Comment:
That whole match should be an extra function, conditional on `send_events` I think